control/server.py

```
@@ -166,7 +168,7 @@ def __exit__(self, exc_type, exc_value, traceback):
             process_name = "discovery"
             if logger:
                 logger.info(f"Exiting the {process_name} process.")
-            return
+            return True
```

does it mean that we hide errors in discovery processes?

---

The discovery process has its own error handling. The only reason we get here on discovery shutdown is that the discovery process was forked from a server process. We don't really want to do anything here for a discovery.